manifests/slave.pp

```diff
@@ -174,13 +174,38 @@
       $defaults_group = 'root'
       $manage_user_home = true
 
-      file { '/etc/init.d/jenkins-slave':
-        ensure => 'file',
-        mode   => '0755',
-        owner  => 'root',
-        group  => 'root',
-        source => "puppet:///modules/${module_name}/jenkins-slave.${::osfamily}",
-        notify => Service['jenkins-slave'],
+      $systemd_min_release = $::osfamily ? {
+        'RedHat' => $::operatingsystem ? {
+          'Fedora' => 21,
+          default  => 7,
+        },
+        default => undef,
+      }
+      if $systemd_min_release {
+        $have_systemd = $::operatingsystemmajrelease >= $systemd_min_release
```

Comment:
If adding such a dependency is acceptable, I'll try to add it, it may make unit testing harder though